### PR TITLE
Fix for layer._dims_displayed removal in napari and update lighting control widget

### PR DESCRIPTION
**Pull request by aganders3**
First of all - this plugin is awesome! I was down the path of changing lighting myself and was very to happy to find this plugin to solve the problem for me.

I am running napari from `main` - I got this error when trying to use the mesh lighting controls:
```
File ~/mf/envs/napari/lib/python3.10/site-packages/napari_threedee/visualization/lighting_control.py:55, in LightingControl._on_camera_change(self=<napari_threedee.visualization.lighting_control.LightingControl object>, event=<Event blocked=False handled=False native=None source=None sources=[] type='angles'>)
     52 view_direction = np.asarray(self._viewer.camera.view_direction)
     54 for layer, visual in zip(self.selected_layers, self.selected_layer_visuals):
---> 55     dims_displayed = layer._dims_displayed
        layer = <Surface layer 'Surface' at 0x2cc27aad0>
     56     layer_view_direction = np.asarray(layer._world_to_data_ray(view_direction))[dims_displayed]
     57     visual.node.shading_filter.light_dir = layer_view_direction[::-1]

AttributeError: 'Surface' object has no attribute '_dims_displayed'
```

This attribute was removed in https://github.com/napari/napari/pull/5003. This PR fixes the issue while maintaining backward-compatibility, and centralizes access in `napari_utils.py` in case it changes again. I also added a few basic tests for `LightingControl` that I think would have caught this.

I also updated the lighting control widget to make things a bit more ergonomic (in my opinion):
* Update the layer list when layers are added/removed, otherwise it was stuck with whatever layers I had when I opened the widget
* Disable the lighting control and button when the viewer is in 2D (adding a tooltip when disabled)

I can remove this part or move it to another PR if you'd like. I haven't checked if other widgets have any similar problems so maybe this could be more generally applied. Of course I'm happy to have feedback or make other changes as well.

### CODE REVIEW

**Comment by alisterburt** [comment]
hey @aganders3 - glad you enjoyed the lighting controls and thanks for the PR, it's much appreciated!

The main fix + tests are really great - the extra things (updating the layer combobox/2d etc) are also great, I'm happy to leave them in here 

cc @kevinyamauchi we should definitely make sure 2D is handled well across the components generally!

Waiting on tests before merge - @kevinyamauchi if you see this before I get back to it and are okay please feel free to merge

**Comment by aganders3** [comment]
Thanks! I was hoping the windows test failure was a fluke, but testing in my fork shows it's repeatable. Unfortunately I don't have great access for testing on Windows but I will look into it.

**Review by kevinyamauchi — APPROVED** [review]
This looks good to me! Thank you @aganders3 !

**Pull request merged**
